PC/winreg.c

```diff
@@ -447,23 +487,21 @@ PyHKEY_AsHKEY(PyObject *ob, HKEY *pHANDLE, BOOL bNoneOK)
 }
 
 BOOL
-clinic_HKEY_converter(PyObject *ob, void *p)
+clinic_HKEY_converter(PyObject *m, PyObject *ob, void *p)
 {
-    if (!PyHKEY_AsHKEY(ob, (HKEY *)p, FALSE))
+    if (!PyHKEY_AsHKEY(m, ob, (HKEY *)p, FALSE))
         return FALSE;
     return TRUE;
 }
 
 PyObject *
-PyHKEY_FromHKEY(HKEY h)
+PyHKEY_FromHKEY(PyObject *m, HKEY h)
 {
-    /* Inline PyObject_New */
-    PyHKEYObject *op = (PyHKEYObject *) PyObject_Malloc(sizeof(PyHKEYObject));
-    if (op == NULL) {
-        return PyErr_NoMemory();
-    }
-    _PyObject_Init((PyObject*)op, &PyHKEY_Type);
+    winreg_state *st = _PyModule_GetState(m);
+
+    PyHKEYObject *op = (PyHKEYObject *) PyObject_GC_New(PyHKEYObject, st->PyHKEY_Type);
```

Comment:
PEP-7 nit

```suggestion
    PyHKEYObject *op = (PyHKEYObject *)PyObject_GC_New(PyHKEYObject,
                                                       st->PyHKEY_Type);
```

Reply:
You also need to check the `PyObject_GC_New` return value, and bail if it failed.